server/src/main/java/io/crate/metadata/upgrade/IndexTemplateUpgrader.java

```diff
@@ -170,4 +178,79 @@ private static void populateColumnPositions(String parentName,
                                     maxColumnPosition);
         }
     }
+
+    // bug : https://github.com/crate/crate/issues/13380
+    private static Map<String, IndexTemplateMetadata> fixCorruptedTemplatesFromSwapTableBug(Map<String, IndexTemplateMetadata> templates) {
+        Map<String, IndexTemplateMetadata> fixedTemplates = new HashMap<>();
+        for (Map.Entry<String, IndexTemplateMetadata> e : templates.entrySet()) {
+            RelationName corruptedRelationName = identifyCorruptedTemplateNames(e.getKey());
+            if (corruptedRelationName != null) {
+                String partnerTemplateName = PartitionName.templateName(corruptedRelationName.schema(), corruptedRelationName.name());
+                IndexTemplateMetadata corruptedTemplate = e.getValue();
+                IndexTemplateMetadata partnerTemplate = templates.get(partnerTemplateName);
+                if (partnerTemplate == null) {
+                    partnerTemplate = fixedTemplates.remove(partnerTemplateName);
+
+                    // unavailable partner template implies that the swap was done on a partitioned and a non-partitioned tables.
+                    // in this case, only the name needs to be fixed
+                    if (partnerTemplate == null) {
+                        fixedTemplates.put(partnerTemplateName, Templates.copyWithNewName(corruptedTemplate, corruptedRelationName).build());
+                        continue;
+                    }
+                }
+
+                RelationName corruptedRelationNameWithBrokenPrefix = new RelationName(
+                    corruptedRelationName.schema(), BROKEN_PREFIX + corruptedRelationName.name());
+
+                // complete the unhandled swap operation
+                IndexTemplateMetadata.Builder corruptedTemplateBuilder = Templates.copyWithNewName(
+                    corruptedTemplate, corruptedRelationNameWithBrokenPrefix);
+                IndexTemplateMetadata.Builder partnerTemplateBuilder = new IndexTemplateMetadata.Builder(partnerTemplate);
+
+                corruptedTemplateBuilder.settings(partnerTemplate.settings());
+                partnerTemplateBuilder.settings(corruptedTemplate.settings());
```

Comment:
This whole method it a bit a hard to understand, I do not completely get the difference of a `partnerTemplate` and a `corruptedTemplate`. Could you maybe add some comments to clarify it?